generatedocs.py

```diff
@@ -0,0 +1,413 @@
+    outputIndex = indexAssetComponentsTemplate.render(assetCategories=assetCategories)
+    with open(assetsOutputPath + '/index.md', 'w') as f:
+        f.write(outputIndex)
+
+    # Print out missing assets
+    total = noOfFoundAssets + len(componentsMissingAssets)
+    percentFound = noOfFoundAssets / (total) * 100
+    print('\n\n')
+    print("Number of found asset examples (ignoring base classes):")
+    print(noOfFoundAssets, "of", total, "or", "%.1f" % percentFound, "%\n")
+    line = "-" * 80 
+    print(line)
+    print(len(componentsMissingAssets), "asset components are missing example files:")
+    print(line) 
+    print('\n'.join(componentsMissingAssets).join("\n\n\n"))
+
+
+################################################################################
+#                            CREATE SCRIPTING API                              #
+################################################################################
+
+def generateScriptingApi(environment, outputFolder, folderNameScripting, jsonLocation):
+    # Create target folder
+    scriptingOutputPath = os.path.join(outputFolder, folderNameScripting)
+    if not os.path.exists(scriptingOutputPath):
+        os.mkdir(scriptingOutputPath)
+
+    # Opening JSON file
+    f = open(os.path.join(jsonLocation, 'scriptingApi.json'))
+
+    # Convert JSON String to Python Dictionary
+    scriptingApi = json.load(f)
+
+    # Create the pages for the scripting libraries
+    scriptingApiTemplate = environment.get_template("scriptingApiTemplate.txt")
+    for library in scriptingApi:
+        # Go through all the functions in that library and print out a md file
+        library = parseDoxygenComments(library)
+        outputLibrary = scriptingApiTemplate.render(library=library)
+        with open(os.path.join(scriptingOutputPath, library["fullName"]+'.md'), 'w') as f:
+            f.write(outputLibrary)
+
+    # Create index file
+    indexScriptingTemplate = environment.get_template("indexScriptingTemplate.txt")
+    outputIndex = indexScriptingTemplate.render(libraries=scriptingApi)
+    with open(os.path.join(scriptingOutputPath, "index.md"), 'w') as f:
+        f.write(outputIndex)
+
+################################################################################
+#                         CREATE RENDERABLE OVERVIEW                           #
+################################################################################
+
+def generateRenderableOverview(environment, outputFolder, folderNameAssets, jsonLocation):
+    # Open JSON file
+    f = open(os.path.join(jsonLocation, 'assetComponents.json'))
+
+    # Convert JSON String to Python Dictionary
+    assetCategories = json.load(f)
+    
+    images = {}
+    renderables = []
+    for category in assetCategories:
+        if category["name"] == "Renderable" or category["name"] == "ScreenSpaceRenderable":
+            for assetComponent in category["classes"]:
+                if assetComponent["name"] == category["name"]:
```

Comment:
Alternate suggestion:

```
                if assetComponent["name"] != category["name"]:
                    # Find example image
                    image = findAssetScreenshot(assetComponent["name"])
                    if image:
                        images[assetComponent["name"]] = image
                    renderables.append(assetComponent)
```

Reply:
I would prefer to not add another layer of nesting